library/kani/src/arbitrary.rs

```diff
@@ -3,15 +3,25 @@
 
 //! This module introduces the Arbitrary trait as well as implementation for primitive types and
 //! other std containers.
+
 use std::{
     marker::{PhantomData, PhantomPinned},
     num::*,
 };
 
 /// This trait should be used to generate symbolic variables that represent any valid value of
 /// its type.
-pub trait Arbitrary {
+pub trait Arbitrary
+where
+    Self: Sized,
+{
     fn any() -> Self;
+    fn any_array<const MAX_ARRAY_LENGTH: usize>() -> [Self; MAX_ARRAY_LENGTH]
+    where
+        [(); std::mem::size_of::<[Self; MAX_ARRAY_LENGTH]>()]:,
```

Comment:
Can you add a comment on what this `where` constraint is doing?

Reply:
I added the following comment
```rust
    // the requirement defined in the where clause must appear on the `impl`'s method `any_array`
    // but also on the corresponding trait's method
```

I also added a comment at the `impl`.
```rust
    // generic_const_exprs requires all potential errors to be reflected in the signature/header.
    // We must repeat the expression in the header, to make sure that if the body can fail the header will also fail.
```